src/v/cluster/members_backend.cc

```diff
@@ -231,15 +251,27 @@ void members_backend::calculate_reallocations_after_decommissioned(
             }
         }
     }
+    co_return;
+}
+
+bool is_reassigned_to_node(
+  const members_backend::partition_reallocation& reallocation,
+  model::node_id node_id) {
+    if (!reallocation.allocation_units.has_value()) {
+        return false;
+    }
+    return is_in_replica_set(
+      reallocation.allocation_units->get_assignments().front().replicas,
+      node_id);
 }
-void members_backend::calculate_reallocations_after_node_added(
-  members_backend::update_meta& meta) const {
+
+ss::future<> members_backend::calculate_reallocations_after_node_added(
```

Comment:
Hmm maybe I am missing something, but what is the point of futurizing these functions? AFAICS they don't co_await anything.

Reply:
true, i will update this in a follow up PR, i have it futurized with maybe_yield in my local branch